Mathlib/Analysis/Convex/Cone/Pointed.lean

```diff
@@ -0,0 +1,189 @@
+/-
+Copyright (c) 2023 Apurva Nakade All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Apurva Nakade
+-/
+import Mathlib.Analysis.Convex.Cone.Dual
+import Mathlib.Algebra.Order.Nonneg.Module
+import Mathlib.Algebra.Module.Submodule.Basic
+
+/-!
+# Pointed cones
+
+A *pointed cone* is defined to be a submodule of a module where the scalars are restricted to be
+nonnegative. This is equivalent to saying that as a set a pointed cone is convex cone which
+contains `0`. This is a bundled version of `ConvexCone.Pointed`. We choose the submodule definition
+as it allows us to use the `Module` API to work with convex cones.
+
+## TODO
+
+- Rewrite proper cones using pointed cones.
+- Construct products and/or direct sums of pointed cones.
+
+-/
+
+variable {𝕜 E F G : Type*}
+
+/-- A pointed cone is a submodule of a module with scalars restricted to being nonnegative. -/
+abbrev PointedCone (𝕜 : Type*) (E : Type*) [OrderedSemiring 𝕜] [AddCommMonoid E] [Module 𝕜 E] :=
+  Submodule { c : 𝕜 // 0 ≤ c } E
+
+namespace PointedCone
+
+-- TODO: remove `prettyPrint := false` once #6833 is merged
+local notation3 (prettyPrint := false) "𝕜≥0" => { c : 𝕜 // 0 ≤ c }
```

Comment:
Can you move that before the `abbrev` and use it there? Also we do not usually put spaces inside curly brackets.
```suggestion
-- TODO: remove `prettyPrint := false` once #6833 is merged
local notation3 (prettyPrint := false) "𝕜≥0" => {c : 𝕜 // 0 ≤ c}
```

Reply:
This doesn't work for some reason. I get the error 

```lean 
failed to synthesize instance
  LE 𝕜✝
```

I used to get the same error when I was trying out `notation` instead of `notation3`

Reply:
What does not work? Using notation in the `abbrev` or removing the spaces?

Reply:
Using notation in abbrev.

Reply:
I've removed the spaces.

Reply:
Would you mind moving the notation earlier, regardless of whether you can use it?

Reply:
Aha, the reason it doesn't work is because you're introducing a new `𝕜` in the definition.

Reply:
I've tried all sorts of permutations but `abbrev` doesn't seem to like `notation`. I'm fixing the duplication of type that you mention. Still the same issue.